src/NimBLEStream.cpp

```
@@ -0,0 +1,934 @@
+        ~Guard() {
+            if (_locked) _b.unlock();
+        }
+        operator bool() const { return _locked; } // Allows: if (Guard g{*this}) { ... }
+    };
+
+    /** @brief Construct a ByteRingBuffer with the specified capacity. */
+    explicit ByteRingBuffer(size_t capacity) : m_capacity(capacity) {
+        memset(&m_mutex, 0, sizeof(m_mutex));
+        auto rc = ble_npl_mutex_init(&m_mutex);
+        if (rc != BLE_NPL_OK) {
+            NIMBLE_LOGE(LOG_TAG, "Failed to initialize ring buffer mutex, error: %d", rc);
```

Cppcheck (2.20.0)

[error] There is an unknown macro here somewhere. Configuration is required. If LOG_RESET_COLOR is a macro then please configure it.

(unknownMacro)